src/main/java/com/minecolonies/colony/ColonyManager.java

```diff
@@ -87,6 +95,36 @@ public static Colony createColony(@NotNull World w, BlockPos pos, @NotNull Entit
     }
 
     /**
+     * Delete a colony and kill all citizens/purge all buildings.
+     *
+     * @param id the colonies id
+     */
+    public static void deleteColony(int id)
+    {
+        final Colony colony = getColony(id);
+        colonies.remove(id);
+        coloniesByWorld.get(colony.getDimension()).remove(colony);
+        final Set<World> colonyWorlds = new HashSet<>();
+        for (CitizenData citizenData : colony.getCitizens().values())
+        {
+            World world = citizenData.getCitizenEntity().getEntityWorld();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'world' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)